spec/unit/Onebip/DateTime/UTCDateTimeTest.php

```diff
@@ -559,4 +559,25 @@ public function testDiff()
                 );
             });
     }
+
+    public function testStartOfMonthWillGiveTheFirstDay()
+    {
+        $this
+            ->forAll(
+                Generator\date(
+                    new DateTime('1980-01-01'),
+                    new DateTime('2020-12-31')
+                )
+            )
+            ->then(function (DateTime $date) {
+                $date->setTimeZone(new DateTimeZone('UTC'));
+                $expected = UTCDateTime::box($date->format('Y-m-01'));
```

Comment:
assertion seems suspiciously duplicated from the production code. What about testing postconditions instead:
- the day must be 1
- the month and the year must be the same as in the input
- the hour, minute, second must be 0